apps/web/src/components/forms/notification-form.tsx

```diff
@@ -46,39 +47,63 @@ interface Props {
   onSubmit?: () => void;
 }
 
+function getDefaultProviderData(defaultValues?: Notification) {
+  if (defaultValues?.provider === "email") {
+    return JSON.parse(defaultValues?.data).email;
+  } else if (defaultValues?.provider === "discord") {
+    return JSON.parse(defaultValues?.data).discord;
+  }
+  return "";
+}
+
+function setProviderData(provider: string, data: string) {
+  let providerJSON = {};
+  switch (provider) {
+    case "email":
+      providerJSON = { email: data };
+      break;
+    case "discord":
+      providerJSON = { discord: data };
+      break;
+    default:
+      providerJSON = {};
+      break;
+  }
+  return providerJSON;
+}
```

Comment:
Adding the type and reducing the complexity. As we `return`, there is no need for `break` statement.

```suggestion
function setProviderData(provider: ProviderName, data: string) {
  switch (provider) {
    case "email":
      return { email: data };
    case "discord":
      return { discord: data };
    default:
      return {};
  }
}
```